### PR TITLE
Add descriptions to some settings.

### PR DESCRIPTION
**Pull request by scorbett123**
This is an example of some settings that I have added the descriptions to, is this the sort of thing that you want doing, if so, are these the correct type of style. There is also a few additions to fix some grammar points, this is also a perfect time to find that type of mistake.

I repeat, if you want this sort of thing to be done then please say, however if not just close this. If you want them I will do more.

### CODE REVIEW

File: src/main/kotlin/org/kamiblue/client/module/modules/chat/AntiSpam.kt [hunk 1]
```diff
@@ -21,14 +21,14 @@ internal object AntiSpam : Module(
...
-    private val slurs = setting("Slurs", true, { page.value == Page.TYPE })
-    private val swears = setting("Swears", false, { page.value == Page.TYPE })
-    private val automated = setting("Automated", true, { page.value == Page.TYPE })
-    private val ips = setting("Server Ips", true, { page.value == Page.TYPE })
-    private val specialCharEnding = setting("Special Ending", true, { page.value == Page.TYPE })
-    private val specialCharBegin = setting("Special Begin", true, { page.value == Page.TYPE })
-    private val greenText = setting("Green Text", false, { page.value == Page.TYPE })
+    private val discordLinks = setting("Discord", true, { page.value == Page.TYPE }, description = "Block messages containing discord links")
+    private val slurs = setting("Slurs", true, { page.value == Page.TYPE }, description = "Removes slurs from messages")
+    private val swears = setting("Swears", false, { page.value == Page.TYPE }, description = "Removes swear words from messages")
```
Comment (l1ving) [inline src/main/kotlin/org/kamiblue/client/module/modules/chat/AntiSpam.kt:26]: I feel these specifically are self explanatory and don't warrant a description, and adding unnecessary descriptions increases the workload for future translations

File: src/main/kotlin/org/kamiblue/client/module/modules/chat/AntiSpam.kt [hunk 2]
```diff
@@ -21,14 +21,14 @@ internal object AntiSpam : Module(
...
+    private val automated = setting("Automated", true, { page.value == Page.TYPE }, description = "Block messages that appear to be automated. E.G. announcer.")
+    private val ips = setting("Server Ips", true, { page.value == Page.TYPE }, description = "Block messages that contain server IP's")
```
Comment (l1ving) [inline src/main/kotlin/org/kamiblue/client/module/modules/chat/AntiSpam.kt:28]: Above comment for the `ips` setting as well

File: src/main/kotlin/org/kamiblue/client/module/modules/chat/AntiSpam.kt [hunk 3]
```diff
@@ -21,14 +21,14 @@ internal object AntiSpam : Module(
...
+    private val specialCharEnding = setting("Special Ending", true, { page.value == Page.TYPE }, description = "Block messages that end with a special character")
+    private val specialCharBegin = setting("Special Beginning", true, { page.value == Page.TYPE }, description = "Block messages that begin with a special character")
+    private val greenText = setting("Green Text", false, { page.value == Page.TYPE }, description = "Block messages continaing green text")
```
Comment (l1ving) [inline src/main/kotlin/org/kamiblue/client/module/modules/chat/AntiSpam.kt:31]: This description would be more helpful if it explained what "green text" is, ie a message that starts with `>`. Also you made a typo in the word containing

File: src/main/kotlin/org/kamiblue/client/module/modules/chat/AntiSpam.kt [hunk 4]
```diff
@@ -38,7 +38,7 @@ internal object AntiSpam : Module(
     private val filterOwn = setting("Filter Own", false, { page.value == Page.SETTINGS })
     private val filterDMs = setting("Filter DMs", false, { page.value == Page.SETTINGS })
     private val filterServer = setting("Filter Server", false, { page.value == Page.SETTINGS })
-    private val showBlocked = setting("Show Blocked", ShowBlocked.LOG_FILE, { page.value == Page.SETTINGS })
+    private val showBlocked = setting("Show Blocked", ShowBlocked.LOG_FILE, { page.value == Page.SETTINGS }, description = "Where to show blocked messages")
```
Comment (l1ving) [inline src/main/kotlin/org/kamiblue/client/module/modules/chat/AntiSpam.kt:41]: This description is unnecessary

File: src/main/kotlin/org/kamiblue/client/module/modules/chat/AutoExcuse.kt [hunk 5]
```diff
@@ -18,10 +18,10 @@ internal object AutoExcuse : Module(
     category = Category.CHAT,
     modulePriority = 500
 ) {
-    private val mode by setting("Mode", Mode.INTERNAL)
+    private val mode by setting("Mode", Mode.PRESET, description = "Where to load excuses from")
```
Comment (l1ving) [inline src/main/kotlin/org/kamiblue/client/module/modules/chat/AutoExcuse.kt:21]: This description is also unnecessary imo

File: src/main/kotlin/org/kamiblue/client/module/modules/chat/AutoQMain.kt [hunk 6]
```diff
@@ -17,9 +17,9 @@ internal object AutoQMain : Module(
     category = Category.CHAT,
     showOnArray = false
 ) {
-    private val showWarns = setting("Show Warnings", true)
-    private val dimensionWarning = setting("Dimension Warning", true)
-    private val delay = setting("Delay", 30, 5..120, 5)
+    private val showWarns = setting("Show Warnings", true, description = "Sends warnings to chat when something is wrong")
+    private val dimensionWarning = setting("Dimension Warning", true, visibility = { showWarns.value }, description = "Warns you when you are in the wrong dimension")
+    private val delay = setting("Delay", 30, 5..120, 5, description = "How freqently to send the message")
```
Comment (l1ving) [inline src/main/kotlin/org/kamiblue/client/module/modules/chat/AutoQMain.kt:22]: Typo in frequently, and I'd deem all of these descriptions unnecessary honestly

File: src/main/kotlin/org/kamiblue/client/module/modules/chat/FancyChat.kt [hunk 7]
```diff
@@ -14,14 +14,14 @@ internal object FancyChat : Module(
...
-    private val uwu = setting("uwu", true)
-    private val leet = setting("1337", false)
-    private val mock = setting("mOcK", false)
-    private val green = setting(">", false)
-    private val blue = setting("`", false)
-    private val randomSetting = setting("Random Case", true, { mock.value })
-    private val commands = setting("Commands", false)
-    private val spammer = setting("Spammer", false)
+    private val uwu = setting("uwu", false, description = "Uwufy all of your messages")
+    private val leet = setting("1337", false, description = "Exchange certain characters for other similar looking characters")
```
Comment (l1ving) [inline src/main/kotlin/org/kamiblue/client/module/modules/chat/FancyChat.kt:18]: Neither of these are really useful. `uwufy` is only useful if the person already understands what `uwu` means in this context

File: src/main/kotlin/org/kamiblue/client/module/modules/chat/FancyChat.kt [hunk 8]
```diff
@@ -14,14 +14,14 @@ internal object FancyChat : Module(
...
+    private val mock = setting("mOcK", false, description = "Change from lower case to upper case")
```
Comment (l1ving) [inline src/main/kotlin/org/kamiblue/client/module/modules/chat/FancyChat.kt:19]: This description would imply it changes all the letters to upper case, which, in reality, it changes letters to a random case

File: src/main/kotlin/org/kamiblue/client/module/modules/chat/FancyChat.kt [hunk 9]
```diff
@@ -14,14 +14,14 @@ internal object FancyChat : Module(
...
+    private val green = setting(">", false, description = "Send all messages as green text")
+    private val blue = setting("`", false, description = "Send all messages as blue text")
```
Comment (l1ving) [inline src/main/kotlin/org/kamiblue/client/module/modules/chat/FancyChat.kt:21]: Again, explain what green / blue text is. I think it's unnecessary when the setting name just matches what it adds

File: src/main/kotlin/org/kamiblue/client/module/modules/chat/FancyChat.kt [hunk 10]
```diff
@@ -14,14 +14,14 @@ internal object FancyChat : Module(
...
+    private val randomSetting = setting("Random Case", true, { mock.value }, description = "Randomize when m0ck happens.")
+    private val commands = setting("Commands", false, description = "Detect commands")
+    private val spammer = setting("Spammer", false, description = "Detect messages sent by spammer")
```
Comment (l1ving) [inline src/main/kotlin/org/kamiblue/client/module/modules/chat/FancyChat.kt:24]: `detect commands` would be confusing and almost incorrect in this case, instead it should be something like `Affect commands`, which could just be the setting name instead.

File: src/main/kotlin/org/kamiblue/client/module/modules/chat/FriendHighlight.kt [hunk 11]
```diff
@@ -18,9 +18,9 @@ internal object FriendHighlight : Module(
     category = Category.CHAT,
     showOnArray = false
 ) {
-    private val bold = setting("Bold", true)
-    private val color = setting("Color", EnumTextColor.GRAY)
-    private val sound = setting("Sound", true)
+    private val bold = setting("Bold", true, description = "Display their name in bold")
+    private val color = setting("Color", EnumTextColor.GRAY, description = "Change the colour of their name")
+    private val sound = setting("Sound", true, description = "Play a sound")
```
Comment (l1ving) [inline src/main/kotlin/org/kamiblue/client/module/modules/chat/FriendHighlight.kt:23]: All of these are unnecessary, and `colour` is not spelled in line with the rest of KAMI Blue (American English)

File: src/main/kotlin/org/kamiblue/client/module/modules/chat/RemoteCommand.kt [hunk 12]
```diff
@@ -18,7 +18,7 @@ internal object RemoteCommand : Module(
     description = "Allow trusted players to send commands",
     category = Category.CHAT
 ) {
-    private val allow = setting("Allow", Allow.FRIENDS)
+    private val allow = setting("Allow", Allow.FRIENDS, description = "Who do you want to be able to see your messages.")
```
Comment (l1ving) [inline src/main/kotlin/org/kamiblue/client/module/modules/chat/RemoteCommand.kt:21]: `Who do you want to be able to see your messages.` really doesn't make sense in this context. The setting is to allow people to send commands to you, not see your messages.

**Review by l1ving — CHANGES_REQUESTED** [review]

**Comment by l1ving** [comment]
I'm closing this as most of the changes were less helpful than helpful. Setting descriptions can be added on a per case basis as needed.

**Comment by scorbett123** [comment]
Ok,  thanks, was just wondering whether this was the sort of thing you wanted.

**Pull request closed without merging**